api/proto/teleport/okta/v1/okta_service.proto

```diff
@@ -51,8 +51,166 @@ service OktaService {
   rpc DeleteOktaAssignment(DeleteOktaAssignmentRequest) returns (google.protobuf.Empty);
   // DeleteAllOktaAssignments removes all Okta assignments.
   rpc DeleteAllOktaAssignments(DeleteAllOktaAssignmentsRequest) returns (google.protobuf.Empty);
+
+  // ValidateClientCredentials checks if the provided client credentials are valid.
+  rpc ValidateClientCredentials(ValidateClientCredentialsRequest) returns (ValidateClientCredentialsResponse);
+  // EnrollIntegration enrolls a new integration with the specified parameters.
+  rpc EnrollIntegration(EnrollIntegrationRequest) returns (EnrollIntegrationResponse);
+  // UpdateIntegration updates the settings or properties of an existing integration.
+  rpc UpdateIntegration(UpdateIntegrationRequest) returns (UpdateIntegrationResponse);
+  // FilterApps retrieves a list of apps based on specified filter criteria.
+  rpc FilterApps(FilterAppsRequest) returns (FilterAppsResponse);
+  // FilterGroups retrieves a list of groups based on specified filter criteria.
+  rpc FilterGroups(FilterGroupsRequest) returns (FilterGroupsResponse);
```

Comment:
Similar to above, should these be just named `GetApps` and `GetGroups`? Let's not invent new terms in this API.

Also, what do these actually do? It's not super clear (to me at least) where these are retrieving apps/groups from, from Okta or from Teleport? Can you expand on that in the godocs please.